source/operators/translate/start_translate.py

```diff
@@ -17,75 +19,129 @@
+    try:
+        asset_id = operator_object.asset_id
+    except KeyError:
+        print('No asset id for this workflow')
+        asset_id = ''
 
-        try:
-            source_lang = operator_object.configuration["SourceLanguageCode"]
-            target_lang = operator_object.configuration["TargetLanguageCode"]
-        except KeyError:
-            operator_object.update_workflow_status("Error")
-            operator_object.add_workflow_metadata(TranslateError="Language codes are not defined")
-            raise MasExecutionError(operator_object.return_output_object())
+    try:
+        source_lang = operator_object.configuration["SourceLanguageCode"]
+        target_lang = operator_object.configuration["TargetLanguageCode"]
+    except KeyError:
+        operator_object.update_workflow_status("Error")
+        operator_object.add_workflow_metadata(TranslateError="Language codes are not defined")
+        raise MasExecutionError(operator_object.return_output_object())
 
-        try:
-            s3_response = s3.get_object(Bucket=bucket, Key=key)
-            transcribe_metadata = json.loads(s3_response["Body"].read().decode("utf-8"))
-            transcript = transcribe_metadata["results"]["transcripts"][0]["transcript"]
-        except Exception as e:
-            operator_object.update_workflow_status("Error")
-            operator_object.add_workflow_metadata(TranslateError="Unable to read transcription from S3: {e}".format(e=str(e)))
-            raise MasExecutionError(operator_object.return_output_object())
+    try:
+        s3_response = s3.get_object(Bucket=bucket, Key=key)
+        transcribe_metadata = json.loads(s3_response["Body"].read().decode("utf-8"))
+        transcript = transcribe_metadata["results"]["transcripts"][0]["transcript"]
+    except Exception as e:
+        operator_object.update_workflow_status("Error")
+        operator_object.add_workflow_metadata(TranslateError="Unable to read transcription from S3: {e}".format(e=str(e)))
+        raise MasExecutionError(operator_object.return_output_object())
 
-        try:
-            translation = translate_client.translate_text(
-                Text=transcript,
-                SourceLanguageCode=source_lang,
-                TargetLanguageCode=target_lang
-            )
-        except Exception as e:
-            operator_object.update_workflow_status("Error")
-            operator_object.add_workflow_metadata(TranslateError="Unable to get response from translate: {e}".format(e=str(e)))
-            raise MasExecutionError(operator_object.return_output_object())
-        else:
+    # Tell the NLTK data loader to look for files in /tmp/
+    nltk.data.path.append("/tmp/")
+    # Download NLTK tokenizers to /tmp/
+    # We use /tmp because that's where AWS Lambda provides write access to the local file system.
+    nltk.download('punkt', download_dir='/tmp/')
+    # Create language tokenizer according to user-specified source language.
```

Comment:
Ideally, MIE should support Transcribe outputs (Translate input in this workflow) for any language supported by the Transcribe service.  Several of the Transcibe supported languages are going to default to English tokenization here.  Have you tried this to make sure we get a reasonable behavior?  Here are the current Transcribe languages: Modern Standard Arabic, Chinese Mandarin-Mainland, Dutch, Australian English, British English, Indian English, Irish English, Scottish English, US English, Welsh English, French, Canadian French, Farsi, German, Indian Hindi, Indonesian, Italian, Korean, Portuguese, Brazilian Portuguese, Russian, Spanish, US Spanish, and Tamil